pinot-core/src/main/java/org/apache/pinot/core/query/distinct/raw/BaseRawIntSingleColumnDistinctExecutor.java

```diff
@@ -29,35 +28,34 @@
 import org.apache.pinot.core.data.table.Record;
 import org.apache.pinot.core.query.distinct.DistinctExecutor;
 import org.apache.pinot.core.query.distinct.DistinctTable;
-import org.apache.pinot.spi.data.FieldSpec.DataType;
+import org.apache.pinot.spi.data.FieldSpec;
 
 
 /**
  * Base implementation of {@link DistinctExecutor} for single raw INT column.
  */
 abstract class BaseRawIntSingleColumnDistinctExecutor implements DistinctExecutor {
   final ExpressionContext _expression;
-  final DataType _dataType;
+  final FieldSpec _fieldSpec;
   final int _limit;
 
-  final IntSet _valueSet;
+  final ObjectSet<Integer> _valueSet;
```

Comment:
This can cause performance regression. We should avoid adding overhead to existing use cases. If we want to return `null` as a separate distinct value, we can add a boolean flag `_hasNull`